packages/controllers/src/subject-metadata/SubjectMetadataController.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  BaseControllerV2 as BaseController,
+  RestrictedControllerMessenger,
+} from '@metamask/controllers';
+import type { Patch } from 'immer';
+import { Json } from 'json-rpc-engine';
+
+import type {
+  GenericPermissionController,
+  PermissionSubjectMetadata,
+  HasPermissions,
+} from '../permissions';
+
+const controllerName = 'SubjectMetadataController';
+
+type SubjectOrigin = string;
+
+export type SubjectMetadata = PermissionSubjectMetadata & {
+  [key: string]: Json;
+  name: string;
+  // TODO:TS4.4 make optional
+  extensionId: string | null;
+  host: string | null;
+  iconUrl: string | null;
```

Comment:
This was called `icon` previously. Was this change intentional?

I suppose this is a better name, but this change will require a migration.

Reply:
Yeah, I'm planning on it!